Tests/RegexTests/AlgorithmsTests.swift

```diff
@@ -114,6 +114,20 @@ class RegexConsumerTests: XCTestCase {
     expectReplace("aab", "a+", "X", "Xb")
     expectReplace("aab", "a*", "X", "XXbX")
   }
+  
+  func testMatches() {
+    let regex = Regex(OneOrMore(.digit).capture { 2 * Int($0)! })
+    let str = "foo 160 bar 99 baz"
+    XCTAssertEqual(str.matches(of: regex).map(\.0.1), [320, 198])
+  }
+  
+  func testMatchReplace() {
+    let regex = Regex(OneOrMore(.digit).capture { Int($0)! })
```

Comment:
It would be great if we get an example of find-and-replace with a regex with multiple captures, both with the regex literal and with the result builder syntax. That will help inform our decisions around Swift tuples vs nominal tuples.